pallets/committee/src/types.rs

```diff
@@ -33,29 +74,31 @@ pub enum CommitteeOrigin<AccountId, BlockNumber> {
 /// Info for keeping track of a motion being voted on.
 /// Default is empty vectors for all votes
 pub struct VoteAggregate<AccountId, BlockNumber> {
-    /// The current set of voters that approved it.
-    pub ayes: Vec<AccountId>,
-    /// The current set of voters that rejected it.
-    pub nays: Vec<AccountId>,
-    /// The current set of votes abstaining.
-    pub abstentions: Vec<AccountId>,
+    /// The current set of votes.
+    pub votes: Vec<MemberVote<AccountId>>,
     /// The hard end time of this vote.
     pub end: BlockNumber,
 }
 
+pub enum VoteRejectionReason {
+    InsuffientVotes,
+    ConstituentVeto,
+    CouncilDeny,
+}
+
 impl<AccountId: Default + PartialEq, BlockNumber: Default> VoteAggregate<AccountId, BlockNumber> {
     pub fn new(
-        ayes: Vec<AccountId>,
-        nays: Vec<AccountId>,
-        abstentions: Vec<AccountId>,
+        ayes: Vec<CommitteeMember<AccountId>>,
+        nays: Vec<CommitteeMember<AccountId>>,
+        abstentions: Vec<CommitteeMember<AccountId>>,
         end: BlockNumber,
     ) -> Self {
-        Self {
-            ayes,
-            nays,
-            abstentions,
-            end,
-        }
+        let votes = std::iter::empty()
+            .chain(ayes.into_iter().map(|x| x.into_vote(Vote::Aye)))
```

Comment:
I think the `std::iter::empty` is not required here

```suggestion
        let votes = ayes.into_iter().map(|x| x.into_vote(Vote::Aye))
```

Reply:
definitely not required but I prefer it for readability. I'm fairly sure it results in the same thing when compiled.

Reply:
You're right it is actually easier to read that way, I like that the relevant iterators are all aligned the same way.